### PR TITLE
Preliminary NixOS rebuilding support

### PR DESCRIPTION
**Pull request by raboof**
Taking pkgs specs of the form:

```
[
  {
    "name": "zlib",
    "version": "1.3@q9jf8xl1gsfmx4f1jhqpdc11yf0w9gv4",
    "distro": "nixos",
    "suite": "core",
    "architecture": "x86_64",
    "input_url": "https://cache.nixos.org/q9jf8xl1gsfmx4f1jhqpdc11yf0w9gv4.narinfo",
    "artifacts": [
      {
        "name": "zlib",
        "version": "1.3",
        "url": "https://cache.nixos.org/nar/1dv9dwz64z347s8b5s78lqrbgdl4xcm0l98zd9cm7jnz1ljd1gjy.nar.xz"
      }
    ]
  }
]
```

Yay:

```
Comparing "/tmp/rebuilderd03zPu6/inputs/1dv9dwz64z347s8b5s78lqrbgdl4xcm0l98zd9cm7jnz1ljd1gjy.nar.xz" with "/tmp/rebuilderd03zPu6/out/1dv9dwz64z347s8b5s78lqrbgdl4xcm0l98zd9cm7jnz1ljd1gjy.nar.xz"
```

### CODE REVIEW

File: worker/Dockerfile.nixos [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+FROM debian:bullseye
+RUN apt-get update && apt install -y libssl-dev git mmdebstrap diffoscope \
+    python3-apt python3-dateutil python3-requests python3-rstr python3-setuptools python3-httpx python3-tenacity \
+    debian-keyring debian-archive-keyring debian-ports-archive-keyring
+# this is a temporary solution
+# copied from https://github.com/fepitre/package-rebuilder/blob/fc38df2f6e81ae6307e47cf515caa3e92ec8f5a4/rebuilder.Dockerfile
+RUN git clone https://salsa.debian.org/python-debian-team/python-debian /opt/python-debian
+RUN cd /opt/python-debian && git checkout e28d7a5729b187cfd0ec95da25030224cd10021a && python3 setup.py build install
+RUN git clone --depth=1 'https://github.com/fepitre/debrebuild' /debrebuild
```
Comment (kpcyrd) [inline worker/Dockerfile.nixos:19]: Most of these you likely want to remove, and maybe even switch the image to a nix based one to skip the installer step below too. :)

File: worker/rebuilder-nixos.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+set -x
+
+echo "Rebuilding $1"
+export PATH=/root/.nix-profile/bin:$PATH
+whoami
+
+DRV=$(cat $1 | grep StorePath | cut -d ":" -f 2)
```
Comment (kpcyrd) [inline worker/rebuilder-nixos.sh:9]:
```suggestion
DRV=$(cat "$1" | grep StorePath | cut -d ":" -f 2)
```
Also most/all of the other variables should also have quotes around them (the unquoted behavior in shell can expand to 0..n words and is very counter intuitive).

**Comment by kpcyrd** [comment]
I'm happy to merge this even though there's currently no scheduler code for NixOS.

**Comment by raboof** [comment]
I'm not sure it makes sense to have 'dead' code in here - I'm also OK to close this PR. I don't remember exactly, but I think I ran into some Nix-specific stuff while preparing this PR, and figured it'd make more sense to start with a Nix-specicific rebuilder and perhaps rolling things back into rebuilderd later. I think that turned into https://github.com/JulienMalka/lila . We can reopen/resume this later.

**Pull request closed without merging**